lxd/instances_post.go

```diff
@@ -1504,16 +1504,22 @@ func clusterCopyContainerInternal(s *state.State, r *http.Request, source instan
 	return createFromMigration(s, nil, projectName, profiles, req)
 }
 
+// instanceCreateFinish finalizes the creation process of an instance by starting it based on
+// the Start field of the request.
 func instanceCreateFinish(s *state.State, req *api.InstancesPost, args db.InstanceArgs) error {
 	if req == nil || !req.Start {
 		return nil
 	}
 
-	// Start the instance.
+	// Load instance and start if not already running.
 	inst, err := instance.LoadByProjectAndName(s, args.Project, args.Name)
 	if err != nil {
 		return fmt.Errorf("Failed to load the instance: %w", err)
 	}
 
+	if inst.IsRunning() {
```

Comment:
This call can be relatively expensive, can we avoid needing this if we just dont try and start it when doing a live/stateful migration?

Reply:
I've updated `createFromMigration` to maintain its current behavior (i.e. not invoking the new `instanceCreateFinish` function), which means we no longer need the `inst.IsRunning` check.